src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/AvoidPotentiallyExpensiveCallWhenLoggingTests.cs

```diff
@@ -0,0 +1,3532 @@
+
+                class C
+                {
+                    void M(ILogger logger, EventId eventId, Exception exception, Func<string, Exception, string> formatter, int input)
+                    {
+                        logger.Log(LogLevel.Debug, eventId, [|$"{input}"|], exception, formatter);
+                    }
+                }
+                """;
+
+            await VerifyCSharpCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        public async Task InvocationOperation_ReportsDiagnostic_CS()
+        {
+            string source = """
+                using System;
+                using Microsoft.Extensions.Logging;
+
+                class C
+                {
+                    void M(ILogger logger, EventId eventId, Exception exception, Func<string, Exception, string> formatter)
+                    {
+                        logger.Log(LogLevel.Debug, eventId, [|exception.ToString()|], exception, formatter);
+                    }
+                }
+                """;
+
+            await VerifyCSharpCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        public async Task IsPatternOperation_ReportsDiagnostic_CS()
+        {
+            string source = """
+                #nullable enable
+
+                using System;
+                using Microsoft.Extensions.Logging;
+
+                class C
+                {
+                    void M(ILogger logger, EventId eventId, Exception? exception, Func<bool, Exception?, string> formatter)
+                    {
+                        logger.Log(LogLevel.Debug, eventId, [|exception is not null|], exception, formatter);
+                    }
+                }
+                """;
+
+            await VerifyCSharpCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        public async Task IsTypeOperation_ReportsDiagnostic_CS()
+        {
+            string source = """
+                using System;
+                using Microsoft.Extensions.Logging;
+
+                class C
+                {
+                    void M(ILogger logger, EventId eventId, Exception exception, Func<bool, Exception, string> formatter, object input)
+                    {
+                        logger.Log(LogLevel.Debug, eventId, [|input is Exception|], exception, formatter);
+                    }
+                }
+                """;
+
+            await VerifyCSharpCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        public async Task NameOfOperation_ReportsDiagnostic_CS()
+        {
+            string source = """
+                using System;
+                using Microsoft.Extensions.Logging;
+
+                class C
+                {
+                    void M(ILogger logger, EventId eventId, Exception exception, Func<string, Exception, string> formatter)
+                    {
+                        logger.Log(LogLevel.Debug, eventId, [|nameof(logger)|], exception, formatter);
```

Comment:
Isn't `nameof` free at runtime?

Reply:
IIRC I used the description from the runtime issue:

>Analyzer would detect when "work" is being done in an argument call to an ILogger logging method. This basically boils down to anything other than a literal-expression, local-expression, parameter-expression, field-expression, property-expression, or indexer-expression (potentially recursively applied).

I think excluding the `nameof` expression is in the spirit of the rule (i.e. no work is done in the log call).